File: app/src/main/java/com/intoverflown/pos/ui/inventory/addproduct/AddProductActivity.java
package com.intoverflown.pos.ui.inventory.addproduct;

import android.content.Intent;
import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

import com.intoverflown.pos.databinding.ActivityAddProductBinding;
import com.intoverflown.pos.ui.inventory.InventoryActivityMain;
import com.intoverflown.pos.utils.Constants;

public class AddProductActivity extends AppCompatActivity {

    private ActivityAddProductBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        binding = ActivityAddProductBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        binding.addBackBtn.setOnClickListener(v -> {
            Intent i = new  Intent(AddProductActivity.this, InventoryActivityMain.class);
            i.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
            startActivity(i);
        });

        binding.addSaveBtn.setOnClickListener(v -> {
            String url = Constants.BASE_URL + "Product/Create";
            postProduct(url);
        });
    }

    private void postProduct(String url) { }
}

File: app/src/main/java/com/intoverflown/pos/ui/inventory/addproduct/AddProductActivity.kt
package com.intoverflown.pos.ui.inventory.addproduct

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.android.volley.toolbox.Volley
import com.intoverflown.pos.databinding.ActivityAddProductBinding
import com.intoverflown.pos.utils.Constants
import org.json.JSONObject

class AddProductActivity : AppCompatActivity() {

    private var binding : ActivityAddProductBinding? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityAddProductBinding.inflate(layoutInflater)
        setContentView(binding!!.root)

//        val uid = SharedPreferenceManager.getInstance(applicationContext).getUser()
//        val token = SharedPreferenceManager.getInstance(applicationContext).getUser().Token
//        Log.d("uid", uid.toString())
//        Log.d("token pref", token.toString())

        binding!!.addBackBtn.setOnClickListener {
//            val i = Intent(this, InventoryActivityMain::class.java)
//            i.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)
//            startActivity(i)
            val url = Constants.BASE_URL + "MerchantBranch/Create"
//            addNewProduct(url, uid.toString(), token.toString())
        }
    }

    private fun addNewProduct(url: String, uid: String, token: String) {
        val queue = Volley.newRequestQueue(applicationContext)
        val jsonObject = JSONObject()


    }
}